File: Examples/Stock Trading Simulation/Go/concreteStrategy.go
package main

import "fmt"

type AggresiveStrategy struct{}
type ConservativeStrategy struct{}

func (a *AggresiveStrategy) DecideAction(stock string, price float64) {
	fmt.Printf("[Aggressive] %s price: %.2f → Action: BUY immediately!\n", stock, price)
}

func (c *ConservativeStrategy) DecideAction(stock string, price float64) {
	if price < 100 {
		fmt.Printf("[Conservative] %s price: %.2f → Action: BUY (undervalued)\n", stock, price)
	} else {
		fmt.Printf("[Conservative] %s price: %.2f → Action: HOLD\n", stock, price)
	}
}

File: Examples/Stock-Trading-Simulation/JS/concreteStrategy.js
import { TradingStrategy } from "./strategyInterface.js";

export class AggressiveStrategy extends TradingStrategy {
  decideAction(stock, price) {
    console.log(`[Aggressive] ${stock} price: ${price} → BUY immediately!`);
  }
}

export class ConservativeStrategy extends TradingStrategy {
  decideAction(stock, price) {
    if (price < 100) {
      console.log(
        `[Conservative] ${stock} price: ${price} → BUY (undervalued)`
      );
    } else {
      console.log(`[Conservative] ${stock} price: ${price} → HOLD`);
    }
  }
}
